addons/mail/static/src/new/discuss/discuss.js

```diff
@@ -92,7 +92,9 @@ export class Discuss extends Component {
     }
 
     get thread() {
-        return this.messaging.state.threads[this.messaging.state.discuss.threadLocalId];
+        return this.props.thread
+            ? this.props.thread
+            : this.messaging.state.threads[this.messaging.state.discuss.threadLocalId];
```

Comment:
Best to just set thread in `messaging.discuss.threadLocalId` in public discuss too